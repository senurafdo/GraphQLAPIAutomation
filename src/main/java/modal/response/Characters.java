package modal.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Characters {
    private Info info;
    private List<Result> results;
}
