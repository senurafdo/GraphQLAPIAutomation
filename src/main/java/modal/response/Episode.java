package modal.response;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Episode {
    private String id;
}
