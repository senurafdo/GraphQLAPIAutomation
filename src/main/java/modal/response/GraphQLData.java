package modal.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class GraphQLData {
    private Characters characters;
    private Location location;
    private List<Episode> episodesByIds;
}
