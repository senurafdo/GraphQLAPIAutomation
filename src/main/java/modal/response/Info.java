package modal.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Info {
    private int count;
}
