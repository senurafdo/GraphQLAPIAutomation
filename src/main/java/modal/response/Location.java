package modal.response;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Location {
    private String id;
}
