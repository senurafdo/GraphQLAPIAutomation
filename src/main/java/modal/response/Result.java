package modal.response;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Result {
    private String name;
}
