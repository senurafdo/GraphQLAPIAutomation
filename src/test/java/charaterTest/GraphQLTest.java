package charaterTest;

import io.restassured.RestAssured;
import io.restassured.builder.RequestSpecBuilder;
import io.restassured.specification.RequestSpecification;
import modal.response.GraphQLGetResponse;
import org.junit.BeforeClass;
import org.junit.Test;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;

public class GraphQLTest {

    @BeforeClass
    public static void beforeAll() {
        start();
        // Set the base URI and base path for the GraphQL endpoint
        RestAssured.baseURI = "https://rickandmortyapi.com";
        RestAssured.basePath = "/graphql";
    }

    @Test
    public void testGraphQLGetRequestWithValidDetails() {
        String body = "{\"query\":\"query Query {\\r\\n " +
                " characters(page: 2, filter: { name: \\\"Morty\\\" }) {\\r\\n   " +
                " info {\\r\\n      count\\r\\n    }\\r\\n    results {\\r\\n    " +
                "  name\\r\\n    }\\r\\n  }\\r\\n  " +
                "location(id: 1) {\\r\\n    id\\r\\n  }\\r\\n  " +
                "episodesByIds(ids: [1, 2]) {\\r\\n    id\\r\\n  }\\r\\n}\",\"variables\":{}}";

        RequestSpecBuilder reqSpecBuilder = new RequestSpecBuilder();
        reqSpecBuilder.setContentType("application/json");
        reqSpecBuilder.setBody(body);

        RequestSpecification requestSpecification = RestAssured.given(reqSpecBuilder.build());
        GraphQLGetResponse graphResponse = requestSpecification.expect().when().post().as(GraphQLGetResponse.class);

        expect(graphResponse).toMatchSnapshot();
    }
}
