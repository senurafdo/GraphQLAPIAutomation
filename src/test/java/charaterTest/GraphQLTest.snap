charaterTest.GraphQLTest.testGraphQLGetRequestWithValidDetails=[
  {
    "data": {
      "characters": {
        "info": {
          "count": 68
        },
        "results": [
          {
            "name": "Hammerhead Morty"
          },
          {
            "name": "Lawyer Morty"
          },
          {
            "name": "Lizard Morty"
          },
          {
            "name": "Long Sleeved Morty"
          },
          {
            "name": "Mechanical Morty"
          },
          {
            "name": "Morty Mart Manager Morty"
          },
          {
            "name": "Morty Jr."
          },
          {
            "name": "Morty Rick"
          },
          {
            "name": "Morty Smith"
          },
          {
            "name": "Morty K-22"
          },
          {
            "name": "Morty Smith"
          },
          {
            "name": "Mortytown Loco"
          },
          {
            "name": "Robot Morty"
          },
          {
            "name": "Slick Morty"
          },
          {
            "name": "Tortured Morty"
          },
          {
            "name": "Toxic Morty"
          },
          {
            "name": "Trunk Morty"
          },
          {
            "name": "Bearded Morty"
          },
          {
            "name": "Morty Jr's interviewer"
          },
          {
            "name": "Bartender Morty"
          }
        ]
      },
      "location": {
        "id": "1"
      },
      "episodesByIds": [
        {
          "id": "1"
        },
        {
          "id": "2"
        }
      ]
    }
  }
]